### ISSUE TITLE
ElegantIcons module not found when importing semantic.css

### ISSUE THREAD

**Issue by jhegg** [Tripwire/octagon#23]
When trying to use octagon as a component library, I'm getting the following errors in my project's webpack build:

```
Failed to compile.

Error in ./~/react-octagon/lib/styles/semantic.min.css
Module not found: ../../src/styles/fonts/ElegantIcons.woff in ~/myproject/ui/node_modules/react-octagon/lib/styles

 @ ./~/css-loader?importLoaders=1!./~/postcss-loader!./~/react-octagon/lib/styles/semantic.min.css 6:117255-117306

Error in ./~/react-octagon/lib/styles/semantic.min.css
Module not found: ../../src/styles/fonts/ElegantIcons.eot in ~/myproject/ui/node_modules/react-octagon/lib/styles

 @ ./~/css-loader?importLoaders=1!./~/postcss-loader!./~/react-octagon/lib/styles/semantic.min.css 6:117335-117385

Error in ./~/react-octagon/lib/styles/semantic.min.css
Module not found: ../../src/styles/fonts/ElegantIcons.ttf in ~/myproject/ui/node_modules/react-octagon/lib/styles

 @ ./~/css-loader?importLoaders=1!./~/postcss-loader!./~/react-octagon/lib/styles/semantic.min.css 6:117427-117477

Error in ./~/react-octagon/lib/styles/semantic.min.css
Module not found: ../../src/styles/fonts/ElegantIcons.svg in ~/myproject/ui/node_modules/react-octagon/lib/styles

 @ ./~/css-loader?importLoaders=1!./~/postcss-loader!./~/react-octagon/lib/styles/semantic.min.css 6:117510-117560
```

In fact, I don't see the ElegantIcons files anywhere in the octagon published npm module. Should they be set as a package dependency?

**Comment by cdaringe** [comment]
there are some fonts in the tripwire theme... maybe those are them?

**Comment by cdaringe** [comment]
hmm, i thought i saw them before. guess not? xref: https://github.com/Tripwire/octagon/pull/18/files

**Comment by eddier** [comment]
When SUI is referencing the Elegant Icons its using the original location for those fonts which is /src/styles/fonts.   Chris and I had talked about including the fonts as a package dependency however we haven't done that yet.

**Comment by jhegg** [comment]
Cool. I can't update the version of Octagon that I'm using until that happens. :smile:

**Comment by eddier** [comment]
Its in progress now, I'll have it for you shortly.

**Comment by jhegg** [comment]
Thank you!

**Comment by jhegg** [comment]
When I use `npm link`, and try to use something like this in my project:
```js
  <Icon className='ei' name='icon_error-triangle_alt' />
```

It doesn't find the ElegantIcon font and instead falls back to my default font, and also gives a warning of:
```
  Warning: Failed prop type: Invalid prop `name` of value `icon_error-triangle_alt` supplied to `Icon`.
```

If I reach across the linked module to `import '../node_modules/react-octagon/src/styles/app.css'`, then the correct font is used (but I still get the invalid prop warning).

So, two questions:
1. Should the `app.css` be distributed somewhere under `react-octagon/lib/*`? Or rather, how should an application that is using react-octagon get the non-default styling/theme? I could just be using it wrong!
2. What can we do about the invalid prop value warning?

**Comment by eddier** [comment]
I'll start with the second issue - the prop name is validated against a predefined set of valid inputs held in an array.   The recommended way around this is to use the className so in the example above it would be transformed as follows:

`<Icon className='ei' name='icon_error-triangle_alt' />`
to
`<Icon className='ei icon_error-triangle_alt' />`

There is a bit of a discussion on this here: https://github.com/Semantic-Org/Semantic-UI-React/issues/931

**Comment by eddier** [comment]
For the first issue yes you probably want to update the builder.js to include a step for the /src/style dir.  @cdaringe how do you want to handle this?

**Comment by cdaringe** [comment]
help me get caught up here.  is the issue that the build process will need to manually move some font assets?

**Comment by eddier** [comment]
Yes the assets located in /src/styles are not available in the /lib dir after the build process is complete.  The current build script handles components and sui, but not the style (fonts & css) objects.

**Comment by cdaringe** [comment]
closed by #25
